Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java

```diff
@@ -0,0 +1,191 @@
+package org.droidplanner.android.fragments.geotag;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.LocalBroadcastManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.animation.Animation;
+import android.view.animation.AnimationUtils;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import org.droidplanner.android.R;
+import org.droidplanner.android.activities.GeoTagActivity;
+
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * Created by chavi on 10/15/15.
+ */
+public class GeoTagImagesFragment extends Fragment {
+    private static final int STATE_INIT = 0;
+    private static final int STATE_GEOTAGGING = 1;
+
+    private GeoTagActivity activity;
+    private LocalBroadcastManager lbm;
+
+    private TextView instructionText;
+    private Button geotagButton;
+    private ProgressBar progressBar;
+    private ImageView sdCard;
+    private Animation sdCardAnim;
+
+    private static final IntentFilter filter = new IntentFilter();
+    static {
+        filter.addAction(GeoTagImagesService.STATE_FINISHED_GEOTAGGING);
+        filter.addAction(GeoTagImagesService.STATE_PROGRESS_UPDATE_GEOTAGGING);
+    }
+
+    private final BroadcastReceiver receiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            switch (intent.getAction()) {
+                case GeoTagImagesService.STATE_FINISHED_GEOTAGGING:
+                    finishedGeotagging(intent);
+                    break;
+                case GeoTagImagesService.STATE_PROGRESS_UPDATE_GEOTAGGING:
+                    int total = intent.getIntExtra(GeoTagImagesService.EXTRA_TOTAL, 0);
+                    int progress = intent.getIntExtra(GeoTagImagesService.EXTRA_PROGRESS, 0);
+                    updateState(STATE_GEOTAGGING);
+                    progressBar.setMax(total);
+                    progressBar.setProgress(progress);
+                    break;
+            }
+        }
+    };
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+
+        if (!(context instanceof GeoTagActivity)) {
```

Comment:
@chaviw should use `getActivity()` here instead of `context`. There's no guarantee that the given `context` is derived from the parent activity.